Automattic/jetpack
# Widget Visibility: Apply widget filtering to customizer preview

The customizer screen was not applying saved visibility rules for
widgets in the preview on the right hand side. This change addresses
that by distinguishing between the customizer and the preview using the
`customize_changeset_uuid` GET var.

Note that when widget visibility rules are first added they will not be
respected until the preview is reloaded, e.g. by navigating inside the
preview or by refreshing page.





Fixes #21446

#### Changes proposed in this Pull Request:

* Preview part of Customizer screen will apply saved widget visibility rules when loaded

#### Does this pull request change what data or activity we track or use?
No.


#### Testing instructions:






* Go to /wp-admin/customize.php
* Press 'Widgets'
* Add a widget
* Use the 'three dots' menu to 'Show more settings'
* Scroll down to 'Advanced'
* Add a visibility rule like 'Show this block if page is about'
* Notice the widget disappear
* In the previewer, navigate to 'About', notice that the visibility rule is applied and you can see the widget
* In the previewer, navigate to any other page, notice that the visibility rule is applied and you cannot see the widget
* Refresh the whole page, notice that the visibility rule is applied and you cannot see the widget

Install and activate the [classic widgets](https://wordpress.org/plugins/classic-widgets/) plugin, and repeat the above. 

Should be tested again on wp.com by applying patch to your sandbox.
